client/my-sites/post-type-list/post-actions-ellipsis-menu/index.jsx

```diff
@@ -15,44 +16,58 @@ import PostActionsEllipsisMenuTrash from './trash';
 import PostActionsEllipsisMenuView from './view';
 import PostActionsEllipsisMenuRestore from './restore';
 
-export default function PostActionsEllipsisMenu( { globalId, includeDefaultActions, children } ) {
-	let actions = [];
-
-	if ( includeDefaultActions ) {
-		actions.push(
-			<PostActionsEllipsisMenuEdit key="edit" />,
-			<PostActionsEllipsisMenuView key="view" />,
-			<PostActionsEllipsisMenuStats key="stats" />,
-			<PostActionsEllipsisMenuPublish key="publish" />,
-			<PostActionsEllipsisMenuRestore key="restore" />,
-			<PostActionsEllipsisMenuTrash key="trash" />
-		);
+export default class PostActionsEllipsisMenu extends PureComponent {
+	static propTypes = {
+		globalId: PropTypes.string,
+		includeDefaultActions: PropTypes.bool,
+		children: PropTypes.node
+	};
+
+	static defaultProps = {
+		includeDefaultActions: true
+	};
+
+	constructor() {
+		super( ...arguments );
+
+		this.renderChildren = this.renderChildren.bind( this );
 	}
 
-	children = Children.toArray( children );
-	if ( children.length ) {
-		if ( actions.length ) {
-			actions.push( <PopoverMenuSeparator key="separator" /> );
+	renderChildren() {
+		const { globalId, includeDefaultActions } = this.props;
+		let { children } = this.props;
+		let actions = [];
+
+		if ( includeDefaultActions ) {
```

Comment:
What is the purpose of this prop and the special logic around `children`?  We don't appear to be using either of these features - if we can remove them, then this function becomes very simple:

`return Children.toArray( children );`